quill-core/src/test/scala/io/getquill/sources/SourceMacroSpec.scala

```diff
@@ -124,4 +124,14 @@ class SourceMacroSpec extends Spec {
     var db = source(new MirrorSourceConfig(""))
     "db.run(qr1)" must compile
   }
+  
+  "fails if bind value is outside `source.run`" in {
+    val q = {
+      val i = 1
```

Comment:
I don't see a passing test for motivation of this PR.

```scala
def helper(p1: Int, p2: Int): Unit = {
  db.run(query[TestEntity].filter(_.i == p1).delete)
}
```
In other words, what makes `val i = 1` a free variable and `p1` in helper method not?